pkg/monitor/controller_test.go

```diff
@@ -0,0 +1,288 @@
+			controller.controllerMonitor.deleteStaleAgentCRDs()
+			crds, _ := controller.controllerMonitor.client.CrdV1beta1().AntreaAgentInfos().List(context.TODO(), metav1.ListOptions{ResourceVersion: "0"})
+			existingNames := []string{}
+			for _, crd := range crds.Items {
+				existingNames = append(existingNames, crd.Name)
+			}
+			assert.Equal(t, tt.expected, existingNames)
+		})
+
+	}
+
+}
+
+func TestSyncControllerCRD(t *testing.T) {
+	controller := newControllerMonitor([]runtime.Object{})
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	controller.informerFactory.Start(stopCh)
+	controller.crdInformerFactory.Start(stopCh)
+	controller.informerFactory.WaitForCacheSync(stopCh)
+	controller.crdInformerFactory.WaitForCacheSync(stopCh)
+	tc := []struct {
+		name          string
+		controllerCRD string
+	}{
+		{
+			name:          "ControllerCRD does not exist",
+			controllerCRD: "",
+		},
+		{
+			name:          "ControllerCRD exists and updated partially",
+			controllerCRD: "TestControllerCRD",
+		},
+	}
+	for _, tt := range tc {
+		t.Run("1", func(t *testing.T) {
+			if tt.controllerCRD == "" {
+				controller.controllerMonitor.syncControllerCRD()
+				assert.Equal(t, crdName, controller.controllerMonitor.controllerCRD.Name)
+			} else {
+				controller.controllerMonitor.controllerCRD, _ = controller.controllerMonitor.createControllerCRD(tt.controllerCRD)
+				controller.controllerMonitor.syncControllerCRD()
+				assert.NotNil(t, controller.controllerMonitor.controllerCRD)
+			}
+
+		})
+	}
+
+}
+
+func newControllerMonitor(objects []runtime.Object) *fakeController {
+	client := fake.NewSimpleClientset()
+	crdClient := fakeclientset.NewSimpleClientset(objects...)
+	informerFactory := informers.NewSharedInformerFactory(client, informerDefaultResync)
+	crdInformerFactory := crdinformers.NewSharedInformerFactory(crdClient, informerDefaultResync)
+	namespaceInformer := informerFactory.Core().V1().Namespaces()
+	serviceInformer := informerFactory.Core().V1().Services()
+	networkPolicyInformer := informerFactory.Networking().V1().NetworkPolicies()
+	nodeInformer := informerFactory.Core().V1().Nodes()
+	cnpInformer := crdInformerFactory.Crd().V1alpha1().ClusterNetworkPolicies()
+	anpInformer := crdInformerFactory.Crd().V1alpha1().NetworkPolicies()
+	tierInformer := crdInformerFactory.Crd().V1alpha1().Tiers()
+	cgInformer := crdInformerFactory.Crd().V1alpha3().ClusterGroups()
+	grpInformer := crdInformerFactory.Crd().V1alpha3().Groups()
+	externalNodeInformer := crdInformerFactory.Crd().V1alpha1().ExternalNodes()
+
+	addressGroupStore := store.NewAddressGroupStore()
+	appliedToGroupStore := store.NewAppliedToGroupStore()
+	networkPolicyStore := store.NewNetworkPolicyStore()
+	groupStore := store.NewGroupStore()
+	groupEntityIndex := grouping.NewGroupEntityIndex()
+	labelIdentityIndex := labelidentity.NewLabelIdentityIndex()
+
+	multiclusterEnabled := features.DefaultFeatureGate.Enabled(features.Multicluster) && true
```

Comment:
I don't think you need this line, you can just pass `false` to  `NewNetworkPolicyController` directly.